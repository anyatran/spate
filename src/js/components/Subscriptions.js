import React from 'react'
import PropTypes from 'prop-types'
import Block from './Block'

const Subscriptions = ({ industry, plan }) => (
  <div className="subscriptions">
    <p className="subscriptions__title">you are subscribed to</p>
    <div className="subscriptions__containers">
      <Block title={plan.label} subtitle={plan.planId} details="See details" url="/"/>
      {Object.keys(industry).map(industryKey => <Block bg={industry[industryKey].image} key={industry[industryKey].key} details={industry[industryKey].label} />)}
    </div>
  </div>
)

Subscriptions.propTypes = {
  industry: PropTypes.object,
  plan: PropTypes.object.isRequired
}

export default Subscriptions;
