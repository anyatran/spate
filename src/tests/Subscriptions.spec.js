import React from 'react'
import { mount } from 'enzyme'
import Subscriptions from '../js/components/Subscriptions'

describe('<Subscriptions />', () => {
  it('should render Subscriptions', () => {
    const wrapper = mount(<Subscriptions industry={{}} plan={{}}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render blocks', () => {
    const plan = {
      lable: 'plan label',
      planId: '123'
    }
    const industry = {
      music: {
        image: 'url',
        key: 'music'
      },
      photography: {
        image: 'url',
        key: 'photography'
      }
    }
    const wrapper = mount(<Subscriptions plan={plan} industry={industry}/>)
    expect(wrapper.find('.block').length).toEqual(3)
  })
})
