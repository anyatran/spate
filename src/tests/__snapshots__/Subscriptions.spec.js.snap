// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subscriptions /> should render Subscriptions 1`] = `
<Subscriptions
  industry={Object {}}
  plan={Object {}}
>
  <div
    className="subscriptions"
  >
    <p
      className="subscriptions__title"
    >
      you are subscribed to
    </p>
    <div
      className="subscriptions__containers"
    >
      <Block
        details="See details"
        url="/"
      >
        <div
          className="block"
        >
          <a
            className="block__container block__container--link"
            href="/"
          >
            <BlockContent
              details="See details"
            >
              <div>
                <p
                  className="block__details"
                >
                  See details
                </p>
              </div>
            </BlockContent>
          </a>
        </div>
      </Block>
    </div>
  </div>
</Subscriptions>
`;
